docs/modules/ROOT/pages/headers.adoc

```diff
@@ -0,0 +1,83 @@
+# DataSonnet Header
+
+The header is an optional comment section in the mapping document which must start with `/\** DataSonnet` followed by a new line character and end with a line `*/` and contains a set of configuration properties.
+The properties are hierarchical, with sections separated by the `.` (dot) character, e.g.:
+
+------------
+input.payload.application/xml.NamespaceSeparator=:
+input.payload.application/xml.TextValueKey=__text
+input.myVar.application/csv.Quote="
+input.myVar.application/csv.Separator=|
+------------
+
+The header must be the first comment before any other line of code, in the following format:
+
+-------------
+/** DataSonnet
+version=<VERSION_NUMBER>
+<PROPERTY>=<VALUE>
+...
+*/
+-------------
+*Example*
+------------------------
+/** DataSonnet
+version=1.0
+*/
+//DataSonnet mapping starts here
+{
+    mappedValue: payload.value
+}
+------------------------
+
+The header is optional; if omitted, all default settings will be used.
+
+## Data Format Properties
+
+The data format properties control the data format plugins behavior. The property names must start with one of `input|output|\*`.
+The input properties format must be `input.<INPUT_NAME>|*.<MIME_TYPE>.<MIME_TYPE_SPECIFIC_PARAMETER>`, where:
+
+    - `<INPUT_NAME>` is a name of input variable or payload;
+    - `<MIME_TYPE>` is a mime type supported by DataSonnet;
+    - `<MIME_TYPE_SPECIFIC_PARAMETER>` is one of the parameters supported by the plugin for this mime type.
+
+If a property is not recognized, an error will be raised.
+
+IMPORTANT: if the mime type contains the dot (`.`) character, it must be escaped with double backslash (`\\`). For example, `dataformat.application/vnd\\.ms-excel.payload.param=xyz`.
+
+The wildcard `*` can be used to set the same parameter value for all inputs and output of the given format, instead of defining it explicitly. For example:
+
+------------
+*.application/csv.Separator=|
```

Comment:
Like we talked about, we should include an XML example with namespaces shared between input and output, which will also illustrate parameters that are maps, and call out specifically how empty strings work (such as for the default namespace).